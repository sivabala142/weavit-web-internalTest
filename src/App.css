@font-face {
  font-family: "DMSans-Bold";
  src: url("./assets/fonts/DMSans-Bold.ttf");
}
@font-face {
  font-family: "DMSans-BoldItalic";
  src: url("./assets/fonts/DMSans-BoldItalic.ttf");
}
@font-face {
  font-family: "DMSans-Italic";
  src: url("./assets/fonts/DMSans-Italic.ttf");
}
@font-face {
  font-family: "DMSans-Medium";
  src: url("./assets/fonts/DMSans-Medium.ttf");
}
@font-face {
  font-family: "DMSans-MediumItalic";
  src: url("./assets/fonts/DMSans-MediumItalic.ttf");
}
@font-face {
  font-family: "DMSans-Regular";
  src: url("./assets/fonts/DMSans-Regular.ttf");
}

html {
  background-color: #fff;
  height: 100%;
}
.App {
  background-color: #fff;
  background-size: cover;
  height: 100%;
}

.MuiTabs-indicator {
  background-color: transparent !important;
}

.css-bcskbc-MuiPaper-root {
  padding-left: 0px !important;
  padding-right: 0px !important;
}
.css-v6wbn9-MuiAvatar-root {
  margin-top: -51px !important;
}
.css-so23x5-MuiPaper-root {
  padding-left: 0px !important;
  padding-right: 0px !important;
}
.MuiPaper-root ::-webkit-scrollbar {
  width: 0;
  background: transparent;
}

.dragged .item {
  opacity: 0;
}


.button-contianer {
  display: flex;
  justify-content: end;
}

.img-container {
  display: flex;
  overflow: hidden;
  width: 100%;
  border: 5px solid #ffffff;
  border-radius: 10px;
}

.dl-list {
  display: flex;
  position: relative;
}

/* tale card avator border */
.backgroundimges {
  background-image: url(./assets/images/top-img-new5.png);
  height: 106px;
  border-radius: 14px;
  background-repeat: no-repeat;
}

/* tree card */
.tree ul {
  margin: 0 0 0 1em;
  padding: 0;
  list-style: none;
  color: #b2b4b8;
  border-width: 12px;
  position: relative;
}
.tree:before,
.tree ul:before {
  content: "";
  display: block;
  width: 0px;
  position: absolute;
  top: 0;
  bottom: 37px;
  left: 8px;
  border-left: 2px solid;
  z-index: -1;
}
.tree li {
  font-weight: bold;
  line-height: 3em;
  margin: 0;
  padding: 1.5em 0 0 1em;
  position: relative;
  z-index: 99;
}
.tree li:before {
  border-top: 2px solid;
  content: "";
  display: block;
  height: 100%;
  left: 8px;
  margin-top: 1em;
  position: absolute;
  top: 2.5em;
  width: 20px;
}
.tree li:last-child:before {
  bottom: 0;
  height: auto;
  top: 1.8em;
}
.tree > li:before {
  border-top: none !important;
}
.tree li div {
  margin: 0;
}

ul.tree {
  margin-left: -44px;
  list-style: none;
}
ul.tree:before {
  border-left: none;
}

.sticky {
  position: -webkit-sticky;
  position: sticky;
  top: 0px;
  width: 100%;
}
.hypertext {
  color: blue;
  text-decoration: underline;
}

.todo {
  display: flex;
}