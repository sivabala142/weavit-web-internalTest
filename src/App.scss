.App {
    display: flex;
}

.bold {
    font-weight: 800;
}
