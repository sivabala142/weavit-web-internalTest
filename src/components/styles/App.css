@font-face {
    font-family: "DMSans-Bold";
    src: url("../fonts/DMSans-Bold.ttf");
  }
  @font-face {
    font-family: "DMSans-BoldItalic";
    src: url("../fonts/DMSans-BoldItalic.ttf");
  }
  @font-face {
    font-family: "DMSans-Italic";
    src: url("./fonts/DMSans-Italic.ttf");
  }
  @font-face {
    font-family: "DMSans-Medium";
    src: url("../fonts/DMSans-Medium.ttf");
  }
  @font-face {
    font-family: "DMSans-MediumItalic";
    src: url("../fonts/DMSans-MediumItalic.ttf");
  }
  @font-face {
    font-family: "DMSans-Regular";
    src: url("../fonts/DMSans-Regular.ttf");
  }
  
  html {
    background-color: #fff;
    height: 100%;
  }
  .App {
    background-color: #fff;
    background-size: cover;
    height: 100%;
  }
  
  .MuiTabs-indicator {
    background-color: transparent !important;
  }
  
  .css-bcskbc-MuiPaper-root {
    padding-left: 0px !important;
    padding-right: 0px !important;
  }
  .css-v6wbn9-MuiAvatar-root {
    margin-top: -51px !important;
  }
  .css-so23x5-MuiPaper-root {
    padding-left: 0px !important;
    padding-right: 0px !important;
  }
  .MuiPaper-root ::-webkit-scrollbar {
    width: 0;
    background: transparent;
  }
  
  /* pagenation */
  
  .pagenationBtts {
    width: 100%;
    list-style: none;
    height: 0px;
    display: flex;
    justify-content: right;
    margin-top: -38px;
  }
  
  .pagenationBtts a {
    padding-top: 15px;
    width: 400px;
    top: 40;
    left: 90%;
    border-radius: 5px;
    font-size: 1px;
    color: "#2b2e2b";
    cursor: pointer;
  }
  
  .pagenationBtts a:hover {
    color: white;
  }
  
  .dragged .item {
    opacity: 0;
  }
  
  .pagenationActive a {
    color: white;
    background-color: #2b2eff;
  }
  
  .button-contianer {
    display: flex;
    justify-content: end;
  }
  
  .img-container {
  display: flex;
    overflow: hidden;
    width: 100%;
    border: 5px solid #ffffff;
    border-radius: 10px;
  }