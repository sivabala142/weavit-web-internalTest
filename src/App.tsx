import React from 'react';
import cookies from 'js-cookie';
import fetch from 'cross-fetch';
import { Routes, Route } from 'react-router-dom';
import { ProtectedRoute } from './layout';
import { GlobalProvider } from './hooks/GlobalContext';
import { CookiesProvider } from 'react-cookie';
import { ApolloProvider, ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client';
import { setContext } from '@apollo/client/link/context';

import { API_BASE_URL } from './config';
import { Dashboard } from './screens/Dashboard';

import { ThemeProvider } from '@mui/material/styles';
import './components/styles/App.css';
import { theme } from './components/theme/default';
import DrawerLayout from './components/Layouts/DrawerLayout';

const httpLink = createHttpLink({
    uri: API_BASE_URL,
    fetch,
});

const authLink = setContext((_, { headers }) => {
    const token = cookies.get('access_token_weavit');
    // return the headers to the context so httpLink can read them
    return {
        headers: {
            ...headers,
            authorization: token ? `Bearer ${token}` : '',
        },
    };
});

const client = new ApolloClient({
    link: authLink.concat(httpLink),
    cache: new InMemoryCache(),
});

function App() {
    return (
        <ThemeProvider theme={theme}>
            <ApolloProvider client={client}>
                <CookiesProvider>
                    <GlobalProvider>
                        <h1>helo</h1>
                        <Routes>
                            <Route path="/" element={<ProtectedRoute />}>
                                <Route element={<DrawerLayout />}>
                                    <Route path="/" element={<Dashboard />} />
                                </Route>
                            </Route>
                        </Routes>
                    </GlobalProvider>
                </CookiesProvider>
            </ApolloProvider>
        </ThemeProvider>
    );
}

export default App;
