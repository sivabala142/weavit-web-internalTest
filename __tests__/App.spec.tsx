import React from 'react';

import App from '../src/App';
import { MemoryRouter } from 'react-router-dom';

import { render } from '@testing-library/react';

jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => 'test'),
            },
        };
    },
}));

global.window = Object.create(window);
const url = 'http://dummy.com';
Object.defineProperty(window, 'location', {
    value: {
        href: url,
    },
    writable: true,
});

describe('render the app', () => {
    it('should render without crashing', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with token expired', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
