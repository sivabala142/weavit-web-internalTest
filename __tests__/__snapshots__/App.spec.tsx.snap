// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the app should render with token expired 1`] = `<DocumentFragment />`;

exports[`render the app should render without crashing 1`] = `<DocumentFragment />`;
